src/tasks/migration/CodeActionProvider.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+import { Commands } from "../../commands";
+import { DiagnosticProvider, diagnosticProvider } from "./DiagnosticProvider";
+import { updateExportTaskType } from "./utils";
+
+export class CodeActionProvider implements vscode.CodeActionProvider {
+
+    public static JAVA_UPDATE_DEPRECATED_TASK_TITLE = "Update deprecated tasks";
+    public static JAVA_UPDATE_DEPRECATED_TASK_FILE_TITLE = "Update deprecated tasks for this file";
+
+    constructor() {
+        vscode.commands.registerCommand(Commands.JAVA_UPDATE_DEPRECATED_TASK, async (filePath) => {
+            await updateExportTaskType(filePath);
+        });
+    }
+
+    public provideCodeActions(document: vscode.TextDocument, range: vscode.Range | vscode.Selection,
+                              _context: vscode.CodeActionContext, _token: vscode.CancellationToken): vscode.CodeAction[] | undefined {
+        const diagnostics = diagnosticProvider.getDiagnosticCollection().get(document.uri);
```

Comment:
We can use `languages.getDiagnostics(document.uri);`, by doing this, we can avoid writing `export const diagnosticProvider = new DiagnosticProvider();` in `DiagnosticProvider.ts`.

// I'm not a fan of export global variable like that since it will introduce a lot of problems when we write unit tests (with webpack enabled).